big-red-shuttle/StopsViewController.swift

```diff
@@ -101,5 +123,186 @@ class StopsViewController: UIViewController, GMSMapViewDelegate {
         let update = GMSCameraUpdate.setTarget(CLLocationCoordinate2DMake(newLat, newLong))
         mapView.animate(with: update)
     }
+    
+    
+    /*This function displays/hides the popUpView based on tapping the marker.*/
+    func mapView(_ mapView: GMSMapView, didTap marker: GMSMarker) -> Bool {
+        if currentMarker == nil {
+            currentMarker = marker
+            popUp(stop: marker.userData as! Stop)
+        } else if currentMarker.position.latitude != marker.position.latitude {
+            dismissPopUpView(marker: marker, fullyDismissed: false)
+        } else {
+            dismissPopUpView(marker: marker, fullyDismissed: true)
+        }
+        return true
+    }
+    
+    
+    func directionsButtonPressed(sender: UIButton) {
+        //prepare to redirect user to map app
+        let location = selectedStop.getLocation()
+        if UIApplication.shared.canOpenURL(URL(string: "comgooglemaps://")!) {
+            if #available(iOS 10.0, *) {
+                UIApplication.shared.open(URL(string:
+                    "comgooglemaps://?saddr=&daddr=\(location.lat),\(location.long)&directionsmode=walking")!)
+            } else {
+                UIApplication.shared.openURL(URL(string:
+                    "comgooglemaps://?saddr=&daddr=\(location.lat),\(location.long)&directionsmode=walking")!)
+            }
```

Comment:
One of them is for ios10+ which is `shared.open` while the other is for ios9 and below `shared.openURL`